streamx-common/src/main/scala/com/streamxhub/streamx/common/fs/LfsOperator.scala

```diff
@@ -87,12 +87,13 @@ object LfsOperator extends FsOperator with Logger {
       val dst = new File(dstPath)
       val inferDstIsDir = {
         if (dst.exists && dst.isDirectory) true
-        else
+        else {
           srcFile.getName -> dst.getName match {
             case (src, dst) if src == dst => false
             case (src, dst) if getSuffix(src) == getSuffix(dst) => false
             case _ => true
           }
+        }
       }
       if (inferDstIsDir) new File(dstPath, srcFile.getName) else dst
     }
```

Comment:
```scala

override def copy(srcPath: String, dstPath: String, delSrc: Boolean, overwrite: Boolean): Unit = {
    if (isAnyBank(srcPath, dstPath)) return

    val srcFile = new File(srcPath)
    if (!srcFile.exists) return
    require(srcFile.isFile, s"[streamx] $srcPath must be a file.")

    def getSuffix(fileName:String) = fileName.drop(fileName.lastIndexOf("."))

    val dstFile = {
      val dst = new File(dstPath)
      val dstWithChild = new File(dstPath, srcFile.getName)
      if (dst.exists && dst.isDirectory) dstWithChild; else {
        (srcFile.getName,dst.getName) match {
            case (s, d) if s == d => dst
            case (s, d) if getSuffix(s) == getSuffix(d) => dst
            case _ => dstWithChild
          }
      }
    }
    val shouldCopy = if (!overwrite && dstFile.exists) false else true

    if (shouldCopy) {
      FileUtils.copyFile(srcFile, dstFile)
      if (delSrc) {
        FileUtils.forceDelete(srcFile)
      }
    }
  }




``` 
is better?

Reply:
I don't quite agree with you that this logic creates a redundant Flie object `dstWithChild` in all cases.

Reply:
> 

I agree with you, it can be improved on this basis